crates/rpc/eth/logs.rs

```diff
@@ -0,0 +1,181 @@
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let address_filter = param.get("address").and_then(|address| match address {
+                    Value::String(_) | Value::Array(_) => {
+                        Some(serde_json::from_value::<AddressFilter>(address.clone()).unwrap())
+                    }
+                    _ => None,
+                });
+                let topics = param.get("topics").map(|topics| {
+                    serde_json::from_value::<Option<Vec<TopicFilter>>>(topics.clone()).unwrap()
+                });
+                Ok(LogsRequest {
+                    from_block,
+                    address_filters: address_filter,
+                    topics: topics.flatten().unwrap_or_else(Vec::new),
+                    to_block,
+                })
+            }
+            _ => Err(RpcErr::BadParams),
+        }
+    }
+    // TODO: This is longer than it has the right to be, maybe we should refactor it.
+    // The main problem here is the layers of indirection needed
+    // to fetch tx and block data for a log rpc response, some ideas here are:
+    // - The ideal one is to have a key-value store BlockNumber -> Log, where the log also stores
+    //   the block hash, transaction hash, transaction number and its own index.
+    // - Another on is the receipt stores the block hash, transaction hash and block number,
+    //   then we simply could retrieve each log from the receipt and add the info
+    //   needed for the RPCLog struct.
+    fn handle(&self, storage: Store) -> Result<Value, RpcErr> {
+        let Ok(Some(from)) = self.from_block.resolve_block_number(&storage) else {
+            return Err(RpcErr::BadParams);
+        };
+        let Ok(Some(to)) = self.to_block.resolve_block_number(&storage) else {
+            return Err(RpcErr::BadParams);
+        };
+
+        let address_filter: BTreeSet<_> = match &self.address_filters {
+            Some(AddressFilter::Single(address)) => std::iter::once(address).collect(),
+            Some(AddressFilter::Many(addresses)) => addresses.iter().collect(),
+            None => BTreeSet::new(),
+        };
+
+        let mut logs: Vec<RpcLog> = Vec::new();
+        // The idea here is to fetch every log and filter by address, if given.
+        // For that, we'll need each block in range, and its transactions,
+        // and for each transaction, we'll need its receipts, which
+        // contain the actual logs we want.
+        for block_num in from..=to {
+            // Take the header of the block, we
+            // will use it to access the transactions.
+            let block_body = storage.get_block_body(block_num)?.ok_or(RpcErr::Internal)?;
+            let block_header = storage
+                .get_block_header(block_num)?
+                .ok_or(RpcErr::Internal)?;
+            let block_hash = block_header.compute_block_hash();
```

Comment:
nit: maybe we want to get it from store (it will be cached so it won't mean a lookup to the db)

Reply:
The Store trait does not provide a method to fetch the hash by number, right? (Or maybe I'm missing something).

Reply:
`get_canonical_block_hash` maybe?